hexrd/imageseries/load/framecache.py

```diff
@@ -127,7 +127,7 @@ def shape(self):
         return self._shape
 
     def __getitem__(self, key):
-        return self._framelist[key].toarray()
+        return self._framelist[key]
```

Comment:
Do we know why `.toarray()` was being used before?

Reply:
Yes! This step actually converts the scipy sparse array into a numpy array. But when this array is used, the only values used are a small subset of the array itself. So, to save on memory usage and run time, I moved this logic to after the subarray is acquired before casting it from sparse array to numpy array.

Previous code:
```python
for i_frame in frame_indices:
    tmp = ome_imgser[i_frame][ijs[0], ijs[1]]
    contains_signal = contains_signal or np.any(
        tmp > threshold
    )
    patch_data_raw.append(tmp)
```

Modified code:
```python
for i_frame in frame_indices:
    tmp = ome_imgser[i_frame][ijs].toarray()
    contains_signal = contains_signal or np.any(
        tmp > threshold
    )
    patch_data_raw.append(tmp)```

Reply:
Nice!

Reply:
However, this changes the return type for one of the image series adapters' `__getitem__` function. I think all of the other image series adapters return a numpy array, is that right @donald-e-boyce? So, for instance, if you load data from an HDF5 file instead, it would be a numpy array, not a scipy sparse array.

Should we keep all of these consistent, @donald-e-boyce?

Reply:
If we should make sure that all image series adapters return a numpy array, could we potentially just call it like this instead, @ZackAttack614?

```
tmp = ome_imgser[i_frame, ijs]
```